### PR TITLE
METRON-548 Improve Profiler documentation

### PR DESCRIPTION
**Pull request by nickwallen**
Reorganized the README of the Profiler to be more sensible, adding links to other reference material like the Profiler Client, and added an additional example showing how the Profiler pairs nicely with Stellar's STATS package.

### CODE REVIEW

File: metron-analytics/metron-profiler/README.md [hunk 1]
```diff
@@ -1,16 +1,74 @@
...
+
+1. Ensure that test messages are being sent to the Profiler's input topic in Kafka.  The Profiler will consume messages from the `inputTopic` defined in the [Profiler's configuration](#configuring-the-profiler).
+
+1. Check the HBase table to validate that the Profiler is writing the profile.  Remember that the Profiler is flushing the profile every 15 minutes.  You will need to wait at least this long to start seeing profile data in HBase.
+    ```
+    $ /usr/hdp/current/hbase-client/bin/hbase shell
+    hbase(main):001:0> count 'profiler'
+    ``` 
+
+1. Use the Profiler Client to read the profile data.  Replace the IP `10.0.0.1` with one specific to your environment.  More information on using the client can be found [here](../metron-profiler-client).
```
Comment (mattf-horton) [inline metron-analytics/metron-profiler/README.md]: Suggest edit:  Use the Profiler Client to read the profile data.  The below example PROFILE_GET command will read data written by the sample profile given above, if `10.0.0.1` is one of the input values for `ip_src_addr`.  More information on using the client can be found [here](../metron-profiler-client).

File: metron-analytics/metron-profiler/README.md [hunk 2]
```diff
@@ -1,16 +1,74 @@
...
+
+1. Stand-up a Metron environment.  For this example, we will use the 'Quick Dev' environment.  Follow the instructions included with [Quick Dev](../../metron-deployment/vagrant/quick-dev-platform) or build your own.
+
+1. Create a table within HBase that will store the profile data. The table name and column family must match the [Profiler's configuration](#configuring-the-profiler).
+    ```
+    $ /usr/hdp/current/hbase-client/bin/hbase shell
+    hbase(main):001:0> create 'profiler', 'P'
+    ```
+    
+1. Define the profile in a file located at `$METRON_HOME/config/zookeeper/profiler.json`.  The following JSON will create a profile that simply counts the number of messages.
```
Comment (mattf-horton) [inline metron-analytics/metron-profiler/README.md]: Suggest edits:
following JSON -> following example JSON
number of messages -> number of messages per ip_src_addr, during each sampling interval.

File: metron-analytics/metron-profiler/README.md [hunk 3]
```diff
@@ -81,21 +139,46 @@ One or more expressions executed when a message is applied to the profile.  A ma
...
+
+```
+$ /usr/metron/0.2.1BETA/start_profiler_topology.sh
+```
+
+## Examples
+
+The following examples are intended to highlight the functionality provided by the Profiler. Each shows the configuration that would be required to generate the profile.  
+
+These examples assume a fictitious input messages that looks something like the following.
```
Comment (mattf-horton) [inline metron-analytics/metron-profiler/README.md]: a fictitious input messages -> a fictitious input message stream

File: metron-analytics/metron-profiler/README.md [hunk 4]
```diff
@@ -210,78 +293,52 @@ This creates a profile...
  * Adds the `length` field from each message
  * Calculates the average as the result
 
-### Topology Configuration
+### Example 4
 
-The Profiler topology also accepts the following configuration settings.
+It is important to note that the Profiler can persist any serializable Object, not just numeric values.  An alternative to the previous example could take advantage of this.
```
Comment (mattf-horton) [inline metron-analytics/metron-profiler/README.md]: This is a very nice new example, showing how to use the stats functions.  Please add a pointer to documentation for those, so the reader can understand what the result data structure 's' looks like.

**Comment by nickwallen** [comment]
Thanks @mattf-horton .  I made the changes that you suggested.

**Comment by cestella** [comment]
+1 by inspection.  I like the stats example

**Pull request closed without merging**